src/shared/protocol.ts

```diff
@@ -360,7 +359,12 @@ export abstract class Protocol<
 
     const responseHandler = this._responseHandlers.get(messageId);
     if (this._timeoutInfo.has(messageId) && responseHandler) {
-      if (!this._resetTimeout(messageId, (reason) => responseHandler(reason as Error))) {
+      try {
+        if (!this._resetTimeout(messageId)) {
+          return;
+        }
```

Comment:
IIUC this means that _not_ having a timeout handler will skip calling the progress handler. Is that intentional? (Sorry for missing it in the first review!)